Merge event-driven-development into main

Duha-Maali opened the pull request:
First version with event-driven task
@yjarrar@asaltech.com

Abdelrahman-Maail committed:
Implement version 1.0 of Event-Driven Application

YazanJarrarAsal commented on EventDrivenApp/Program.cs:
```diff
@@ -1,2 +1,43 @@
+        publisher.NotifyPromotions("50% off on all items!", DateTime.Now);
+        publisher.NotifySocial("You have a new friend request.", DateTime.Now);
+        publisher.NotifyUpdates("This new IDE from Google is an absolute game changer | Tari Ibaba in Coding Beauty", DateTime.Now);
+
```
remove empty spaces

YazanJarrarAsal reviewed (changes requested):
Amazing job but you need to add bin and obj files to gitignore before completing the pull request

Abdelrahman-Maail committed:
fix: add bin/obj to .gitignore and remove empty spaces in Program.cs

Duha-Maali commented:
I’ve made the requested changes:
1: Added the bin/ and obj/ directories to the .gitignore file.
2: Removed empty spaces in Program.cs.

Please let me know if any further changes are needed.
Thanks for your review!
